src/main/java/hudson/plugins/sonar/MsBuildSQRunnerBegin.java

```diff
@@ -0,0 +1,252 @@
+      }
+    }
+
+    args.addTokenized(sonarInst.getAdditionalProperties());
+    args.addTokenized(additionalArguments);
+  }
+
+  @Override
+  public DescriptorImpl getDescriptor() {
+    return (DescriptorImpl) super.getDescriptor();
+  }
+
+  private SonarInstallation getSonarInstallation(TaskListener listener) throws AbortException {
+    if (!SonarInstallation.isValid(getSonarInstallationName(), listener)) {
+      throw new AbortException();
+    }
+    return SonarInstallation.get(getSonarInstallationName());
+  }
+
+  /*
+   * FOR UI
+   */
+  public String getProjectKey() {
+    return Util.fixEmptyAndTrim(projectKey);
+  }
+
+  public String getProjectVersion() {
+    return Util.fixEmptyAndTrim(projectVersion);
+  }
+
+  public String getProjectName() {
+    return Util.fixEmptyAndTrim(projectName);
+  }
+
+  public String getSonarInstallationName() {
+    return Util.fixEmptyAndTrim(sonarInstallationName);
+  }
+
+  public String getMsBuildRunnerInstallationName() {
+    return Util.fixEmptyAndTrim(msBuildRunnerInstallationName);
+  }
+
+  public String getAdditionalArguments() {
+    return Util.fixEmptyAndTrim(additionalArguments);
+  }
+
+  /*
+   * 
+   */
+  @Extension
+  public static class DescriptorImpl extends BuildStepDescriptor<Builder> {
+    @Override
+    public boolean isApplicable(Class<? extends AbstractProject> jobType) {
+      return true;
+    }
+
+    @Override
+    public String getHelpFile() {
+      return "/plugin/sonar/help-ms-build-sq-runner-begin.html";
+    }
+
+    @Override
+    public String getDisplayName() {
+      return Messages.MsBuildRunnerBegin_DisplayName();
+    }
+
+    public FormValidation doCheckProjectKey(@QueryParameter String value) {
+      return checkNotEmpty(value);
+    }
+
+    public FormValidation doCheckProjectName(@QueryParameter String value) {
+      return checkNotEmpty(value);
+    }
+
+    public FormValidation doCheckProjectVersion(@QueryParameter String value) {
+      return checkNotEmpty(value);
+    }
+
+    private static FormValidation checkNotEmpty(String value) {
+      if (!StringUtils.isEmpty(value)) {
+        return FormValidation.ok();
+      }
+      return FormValidation.error(Messages.SonarPublisher_MandatoryProperty());
+    }
+
+    @Nullable
+    public MsBuildSQRunnerInstallation getMsBuildRunnerInstallation(String name) {
+      MsBuildSQRunnerInstallation[] msInst = getMsBuildRunnerInstallations();
+
+      if (name == null && msInst.length > 0) {
+        return msInst[0];
+      }
+
+      for (MsBuildSQRunnerInstallation inst : msInst) {
+        if (StringUtils.equals(name, inst.getName())) {
+          return inst;
+        }
+      }
+
+      return null;
+    }
+
+    public MsBuildSQRunnerInstallation[] getMsBuildRunnerInstallations() {
+      return Jenkins.getInstance().getDescriptorByType(MsBuildSQRunnerInstallation.DescriptorImpl.class).getInstallations();
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) NullPointerException might be thrown as 'getInstance' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2259)